plugins/myMetric/api/api.js

```diff
@@ -0,0 +1,49 @@
+var plugins = require('../../pluginManager.js');
+
+
+plugins.register("/i/my-metric", function (ob) {
+    let paramsInstance = ob.params;
+    if (paramsInstance.qstring.args) {
+        paramsInstance.qstring.args.date = new Date();
+        try {
+            paramsInstance.qstring.args = JSON.parse(paramsInstance.qstring.args);
+        } catch (SyntaxError) {
+            console.log('Parse ' + paramsInstance.qstring.args + ' JSON failed');
+        }
+    }
+    common.db.collection('my_metric').insert(paramsInstance, function () {
+        return true;
+    });
+});
+
+plugins.register("/o/my-metric", function (ob) {
+    var params = ob.params;
+    var validate = ob.validateUserForGlobalAdmin;
+    if (params.qstring && params.qstring.args) {
+        try {
+            params.qstring.args = JSON.parse(params.qstring.args);
+        } catch (SyntaxError) {
+            console.log('Parse ' + paramsInstance.qstring.args + ' JSON failed');
```

Comment:
Codacy found an issue: ['paramsInstance' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6388729)